core/arch/arm/kernel/entry_a32.S

```diff
@@ -571,9 +571,17 @@ shadow_stack_access_ok:
 
 #ifdef _CFG_CORE_STACK_PROTECTOR
 	/* Update stack canary value */
-	bl	plat_get_random_stack_canary
+	mov	r7, sp
+	sub	sp, sp, #0x8
+	bic	sp, sp, #7	/* make sure it's a multiple of 8 */
```

Comment:
This shouldn't be needed, the SP must always be a multiple of 8 when calling functions.